internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -290,5 +285,39 @@ func createWindowsContainerDocument(ctx context.Context, coi *createOptionsInter
 	}
 	v1.MappedPipes = mpsv1
 	v2Container.MappedPipes = mpsv2
+
+	// add assigned devices to the container definition
+	parseAssignedDevices(coi, v2Container)
 	return v1, v2Container, nil
 }
+
+func specHasAssignedDevices(coi *createOptionsInternal) bool {
+	if coi.Spec.Windows != nil && coi.Spec.Windows.Devices != nil &&
+		len(coi.Spec.Windows.Devices) > 0 {
+		return true
+	}
+	return false
+}
+
+// parseAssignedDevices parses assigned devices for the container definition
+// this is currently supported for v2 argon only
+func parseAssignedDevices(coi *createOptionsInternal, v2 *hcsschema.Container) {
+	if !specHasAssignedDevices(coi) || !coi.isV2Argon() {
```

Comment:
Add a comment here explaining why we skip if `isV2Argon` is false.

Reply:
That's in the comment above. This is supported currently only for v2 argon.